src/main/java/com/aws/greengrass/cisclient/ConnectivityInfoProvider.java

```diff
@@ -59,14 +70,19 @@ public List<String> getCachedHostAddresses() {
      * Get connectivity info.
      *
      * @return list of connectivity info items
+     * @throws Exception when not able to retrieve greengrasV2DataClient
```

Comment:
Throw only the `DeviceConfigurationException` exception. Do you have `Exception` because of the `RetryUtils`?

Reply:
yes, RetryUtils required Exception